src/Smalot/PdfParser/Document.php

```diff
@@ -144,9 +149,116 @@ protected function buildDetails()
             $details['Pages'] = 0;
         }
 
+        $details = array_merge($details, $this->metadata);
+
         $this->details = $details;
     }
 
+    /**
+     * Extract XMP Metadata
+     */
+    public function extractXMPMetadata(string $content): void
+    {
+        $xml = xml_parser_create();
+        xml_parser_set_option($xml, \XML_OPTION_SKIP_WHITE, 1);
+
+        if (xml_parse_into_struct($xml, $content, $values, $index)) {
+            $detail = '';
+
+            foreach ($values as $val) {
+                switch ($val['tag']) {
+                    case 'DC:CREATOR':
+                        $detail = ('open' == $val['type']) ? 'Author' : '';
+                        break;
+
+                    case 'DC:DESCRIPTION':
+                        $detail = ('open' == $val['type']) ? 'Description' : '';
+                        break;
+
+                    case 'DC:TITLE':
+                        $detail = ('open' == $val['type']) ? 'Title' : '';
+                        break;
+
+                    case 'DC:SUBJECT':
+                        $detail = ('open' == $val['type']) ? 'Subject' : '';
+                        break;
+
+                    case 'RDF:LI':
+                        if ('' !== $detail && 'complete' == $val['type'] && isset($val['value'])) {
```

Comment:
I replaced `$detail` with `'' !== $detail` in this if-clause. Now its more clear what this part means. The former one was ambiguous, because PHP uses magic sometimes to decide what values related to `true` when using `if($var)`.